### PR TITLE
Fixed Search term minimum length is static

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fixed Search term minimum length is static
### Fixed Issues (if relevant)

1. magento/magento2#12053: Search term minimum length is static

### Manual testing scenarios (*)

1. N/A
### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by miguelbalparda** [comment]
This does not solve the mentioned issue, it's just changing the static value from 3 to 1. I don't see why this should be changed, reason is explained in #12053. Feel free to reopen if you have anything to add or more changes to fix the mentioned issue

**Comment by ravi-chandra3197** [comment]
Hello @miguelbalparda
I have update code can you review it.

**Review by miguelbalparda — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @miguelbalparda, thank you for the review.
[ENGCOM-4398](https://jira.corp.magento.com/browse/ENGCOM-4398) has been created to process this Pull Request

File: lib/internal/Magento/Framework/Search/Adapter/Mysql/Query/Builder/Match.php [hunk 1]
```diff
@@ -73,6 +76,9 @@ public function __construct(
         $this->fulltextHelper = $fulltextHelper;
         $this->fulltextSearchMode = $fulltextSearchMode;
         $this->preprocessors = $preprocessors;
+        $this->_ConfigInterface = $ConfigInterface;
+        $this->minQueryLength = (int)$this->_ConfigInterface->getValue(
```
Comment (sivaschenko) [inline lib/internal/Magento/Framework/Search/Adapter/Mysql/Query/Builder/Match.php:80]: Hi @ravi-chandra3197 I wouldn't reference catalog search configuration path in the framework, can you refactor the fix to move the config reference out of the framework?

**Review by sivaschenko — COMMENTED** [review]
Hi @ravi-chandra3197 thanks for the pull request! Please see my comments review comments

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @ravi-chandra3197 thanks for the pull request! Please see my comments review comments

**Comment by sidolov** [comment]
@ravi-chandra3197 , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**